compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -3655,8 +3662,13 @@ private BLangLiteral createSimpleLiteral(Node literal, SyntaxKind sign) {
         } else if (type == SyntaxKind.DECIMAL_FLOATING_POINT_LITERAL) {
             //TODO: Check effect of mapping negative(-) numbers as unary-expr
             typeTag = NumericLiteralSupport.isDecimalDiscriminated(textValue) ? TypeTags.DECIMAL : TypeTags.FLOAT;
-            value = textValue;
-            originalValue = textValue;
+            if (isFiniteType) {
+                value = textValue.replaceAll("[f,d,/+]", "");
```

Comment:
no need to do replaceAll, which may be slower, also why are there `,` commas between chars.
`. replaceFirst("[fd]$", "");`

Reply:
Oh is this replacing the leading `+` as well? and what about `/` I though escaping is done by `\\`

Reply:
removed comma and /. :-)